src/NServiceBus.Core/Hosting/Helpers/AssemblyScanner.cs

```diff
@@ -395,23 +377,21 @@ static bool ReferencesNServiceBus(Assembly assembly, string coreAssemblyName, Di
                     continue;
                 }
 
-                if (assemblyName.Name == coreAssemblyName)
+                if (referencedAssemblyName.Name == coreAssemblyName)
                 {
                     processed[assemblyNameFullName] = true;
                     return true;
                 }
 
-                if (IsRuntimeAssembly(assemblyName))
+                if (IsRuntimeAssembly(referencedAssemblyName))
                 {
                     processed[assemblyNameFullName] = false;
                     continue;
                 }
-                //We need to do a ApplyPolicy, and use the result, so as to respect the current binding redirects
-                var afterPolicyName = AppDomain.CurrentDomain.ApplyPolicy(assemblyNameFullName);
                 Assembly refAssembly;
                 try
                 {
-                    refAssembly = Assembly.ReflectionOnlyLoad(afterPolicyName);
+                    refAssembly = Assembly.Load(assemblyNameFullName);
```

Comment:
With this change, I wonder if the `catch` blocks below are still valid, or should it be changed to `catch (Exception ex) when (ex is BadImageFormatException || ex is FileLoadException)` like they are elsewhere.